packages/pipeline-editor/src/PipelineEditor/index.tsx

```diff
@@ -635,6 +635,75 @@ const PipelineEditor = forwardRef(
       return null;
     };
 
+    const getUpstreamNodeIds = (node: any): string[] => {
+      let upstreamNodeIds: string[] = [];
+
+      for (const link of node.inputs?.[0].links || []) {
+        upstreamNodeIds.push(link.node_id_ref);
+        const upstreamNode = controller.current.findExecutionNode(
+          link.node_id_ref
+        );
+        if (upstreamNode) {
+          upstreamNodeIds = upstreamNodeIds.concat(
+            getUpstreamNodeIds(upstreamNode)
+          );
+        }
+      }
+
+      return upstreamNodeIds;
+    };
+
+    const getNodeProperties = (selectedNode: any): any => {
+      const data: any[] = [];
+
+      const nodes = controller.current.getAllPaletteNodes();
+
+      const upstreamNodes = controller.current.idsToNodes(
+        getUpstreamNodeIds(selectedNode)
+      );
+
+      for (const upstreamNode of upstreamNodes) {
+        // @ts-ignore
```

Comment:
Why are we adding this ignore?

Reply:
we have to have the ignore because we're accessing `op` on the selected node, but as far as the compiler knows the node could be a super node which doesn't have the property `op`. This case can't happen though because this function is called  after a node type check to make sure it's an execution node. @bourdakos1 if there's a better way to handle this I'm open to it.

Reply:
we should definitely avoid the ts-ignore. There are a couple ways we can get around it. The easiest is just casting upstreamNodes to a list of execution nodes, but there is probably a better solution

Reply:
I think my comment disappeared...? but yea, we should avoid ts-ignore. Easiest way is to just cast upstreamNodes to a list of execution nodes, but that is kind of a bandaid and there is probably a better long term solution

Reply:
I did a little bit of googling into the best ways to handle type assertion in typescript and the method that seem to match our use case the best is to cast `(upstreamNode as ExecutionNodeDef)?.op` inline. This cast will be `undefined` if the type doesn't match, which in turn would fail the check in the find.

Reply:
I've pushed the above fix, feel free to review again